src/components/Carousel/index.tsx

```diff
@@ -0,0 +1,344 @@
+import type { ReactElement, TouchEventHandler } from 'react'
+import { useEffect, useState } from 'react'
+import { ICON } from '@constants/icons'
+import styled from '@emotion/styled'
+import useMediaQuery from 'hook/useMediaQuery'
+
+type GetMediaQuery = (breakPoint: keyof typeof BREAKE_POINTS) => string
+const BREAKE_POINTS = {
+  DESKTOP: 1920,
+  MOBILE: 699,
+  TABLET: 1023
+} as const
+
+const getMediaQuery: GetMediaQuery = breakpoint => {
+  switch (breakpoint) {
+    case 'DESKTOP':
+      return `@media (max-width: ${BREAKE_POINTS.DESKTOP}px)`
+    case 'MOBILE':
+      return `@media (max-width: ${BREAKE_POINTS.MOBILE}px)`
+    case 'TABLET':
+      return `@media (max-width: ${BREAKE_POINTS.TABLET}px)`
+    default:
+      return ``
+  }
+}
+
+export interface CarouselProps {
+  images: { url: string; id: number }[]
+  isArrow: boolean
+  size?: number
+  name: string
+}
+
+interface SliderProps {
+  cursorOn: boolean
+  size: number
+}
+
+interface CurrentIndicatorProps {
+  imageIndex: number
+}
+
+interface ImageBoxProps {
+  translateValue: number | null
+}
+
+interface ImageProps {
+  size: number
+}
+
+interface ArrowProps {
+  translateValue: number | null
+}
+
+interface IndicatorBoxProps {
+  isArrow: boolean
+}
+
+type HandleOffset = (navType: keyof typeof NAV_TYPE) => void
+const NAV_TYPE = {
+  LEFT: 'LEFT',
+  RIGHT: 'RIGHT'
+} as const
+
+const Carousel = ({
+  images,
+  isArrow,
+  size = 687,
+  name
+}: CarouselProps): ReactElement => {
+  const desktop = useMediaQuery(`(min-width:1023px)`)
+
+  const carouselWidthSize = desktop ? size : 100
+  const translateValueOfLastImage = carouselWidthSize * (images.length - 1)
+  const arrowRight = ICON.CHEVRON_RIGHT_40
+  const arrowLeft = ICON.CHEVRON_LEFT_40
+  const [translateValue, setTranslateValue] = useState<number>(0)
```

Comment:
translateValue는 현재 선택된 이미지 값을 의미하는데 
translateValue는 변경한 값이라는 뜻을 가졌다고 생각해서 의미가 모호하다는 느낌이 들었어용
더 직관적인 변수명을 고민해보는건 어떨까요 ?

제가 지금 바로 생각나는 거는 요정도입니다 ㅎㅎ 

```suggestion
  const lastImageValue = carouselWidthSize * (images.length - 1)
  const arrowRight = ICON.CHEVRON_RIGHT_40
  const arrowLeft = ICON.CHEVRON_LEFT_40
  const [currentImageValue, setCurrentImageValue] = useState<number>(0)
```

또 이미지 src를 변수로 선언하셨는데 사용하는 곳은 image 적용하는 한곳 이더라구요
제가 추측한바로는 추후 변경이 쉽도록 상수화 작업을 하기 위해서 변수를 선언하신 것 같아요!
혹시 맞다면 최상단에 대문자로 선언하는건 어떨지 의견 내봅니다 ㅎㅎ